scripts/Diffraction/isis_powder/abstract_inst.py

```diff
@@ -69,6 +69,11 @@ def set_beam_parameters(self, height, width):
         :param width: Width of the beam (mm).
         :return:
         """
+        try:
+            height = float(height)
+            width = float(width)
+        except ValueError:
+                raise ValueError("Beam height and width must be numbers.")
```

Comment:
This takes effect only if the height or width is given as a string e.g. 'h'. If instead one put in h, one get h undefined error and if one puts in 1h one get syntax error. I think this is a feature of python and so can't be avoided.